themes/default/content/blog/kubernetes-v4-shared-resource-management-with-server-side-apply/index.md

```diff
@@ -0,0 +1,219 @@
+---
+title: "Kubernetes 4.0: Shared Resource Management With Server-Side Apply"
+
+date: 2023-07-19T09:00:00-04:00
+
+meta_desc: The Pulumi Kubernetes Provider 4.0; Shared Resource Management with Server-Side Apply
+
+meta_image: meta.png
+
+authors:
+    - levi-blackstone
+    - luke-hoban
+    - monica-rodriguez
+
+tags:
+    - kubernetes
+    - providers
+
+---
+
+Since the very earliest days of the Pulumi project, Kubernetes has been a core part of the Pulumi platform.  The initial Pulumi Kubernetes provider supported the entire API surface area of the Kubernetes platform, derived directly and automatically from the Kubernetes OpenAPI specifications.  Since then, we have offered day one support for every new Kubernetes version, added support for Helm, YAML, Kustomize and CRDs, added tools for converting to Pulumi (kube2pulumi and crd2pulumi) and delivered the Pulumi Kubernetes Operator.  During that same time, Kubernetes usage has continued to expand within the ecosystem and within Pulumi users, with the Kubernetes provider growing from the fourth most used to the second most used provider on the platform.
```

Comment:
```suggestion
Since the very earliest days of the Pulumi project, Kubernetes has been a core part of the Pulumi platform and has grown from the fourth most used to the second most used provider on the platform. Pulumi simplifies Kubernetes management by allowing you to express your K8s resources with familiar programming languages. Pulumi users love the Kubernetes Native Provider because it provides the most complete and up-to-date coverage of the Kubernetes platform and it natively supports K8s constructs and best practices like CI/CD. The initial Pulumi Kubernetes provider supported the entire API surface area of the Kubernetes platform, derived directly and automatically from the Kubernetes OpenAPI specifications.  Since then, we have offered day-one support for every new Kubernetes version, added support for Helm, YAML, Kustomize and CRDs, added tools for converting to Pulumi (kube2pulumi and crd2pulumi) and delivered the Pulumi Kubernetes Operator.
```

Reply:
rewrote this a bit to position the usage stats upfront (those are most impressive), then recap why Kubernetes Provider is unique and differentiated for our users. i think this will be more impactful while also educating anyone who's not familiar with the provider yet

Reply:
I took a few of these changes into the latest commit, but left some of it as-is.  Open to additional changes if still needed - though I prefer to not lead with the usage stat in this particular case.